src/main/java/shrek/Shrek.java

```diff
@@ -0,0 +1,476 @@
+    }
+
+    public static String revertDatetime(String datetime) throws ParseException {
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd hh:mm");
+        Date date = new SimpleDateFormat("MMM d yyyy hh:mma").parse(datetime);
+        return formatter.format(date);
+    }
+
+    public static void saveDeadlineToOutput(Deadlines task) throws IOException, ParseException {
+        String baseString = "deadline";
+        String mark = convertMark(task);
+        String taskContent = task.getContent();
+        String taskBy = task.getBy();
+        String userInputStyleDatetime = revertDatetime(taskBy);
+        baseString = mark + " " + baseString + " " + taskContent + "/by " + userInputStyleDatetime;
+        writeToFile(baseString);
+    }
+
+    public static void saveEventToOutput(Events task) throws IOException, ParseException {
+        String baseString = "event";
+        String mark = convertMark(task);
+        String taskContent = task.getContent();
+        String taskAt = task.getAt();
+        String userInputStyleDatetime = revertDatetime(taskAt);
+        baseString = mark + " " + baseString + " " + taskContent + "/at " + userInputStyleDatetime;
+        writeToFile(baseString);
+    }
+
+    public static void writeToFile(String task) throws IOException {
+        FileWriter outputFile = new FileWriter(OUTPUT_FILE_PATH, true);
+        outputFile.write(task + System.lineSeparator());
+        outputFile.close();
+    }
+
+    public static void clearOutput() throws IOException {
+        FileWriter outputFile = new FileWriter(OUTPUT_FILE_PATH, false);
+        outputFile.write("");
+        outputFile.close();
+    }
+
+    public static void findTaskOrTime(String input) {
+        try {
+            String[] splitTaskOrTimeInputs = input.split(" ", NUMBER_OF_TERMS_IN_SPLIT);
+            if (splitTaskOrTimeInputs[FIND_CONTENT].equals("")) {
+                throw new InvalidCommandException("Did you forget to input time or task?", errorCount);
+            }
+            switch (splitTaskOrTimeInputs[FIND_TASK_OR_TIME]) {
+            case "time":
+                findTime(splitTaskOrTimeInputs[FIND_CONTENT]);
+                break;
+            case "task":
+                findTask(splitTaskOrTimeInputs[FIND_CONTENT]);
+                break;
+            default:
+                throw new InvalidCommandException("Did you type time or task wrongly?", errorCount);
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new InvalidCommandException("Did you forget to declare the time or task?", errorCount);
+        }
+    }
+
+    public static void findTask(String inputToSearchFor) {
+        ArrayList<UserContent> listOfFoundTasks = new ArrayList<>();
+        for (UserContent item : lists) {
+            if (item.getContent().contains(inputToSearchFor)) {
+                listOfFoundTasks.add(item);
+            }
+        }
+        if(listOfFoundTasks.size() == 0){
```

Comment:
**Coding standard violation:** Add a whitespace character between `if`, `(listOfFoundTasks.size() == 0)`, and `{`